.github/workflows/build.yml

```diff
@@ -0,0 +1,26 @@
+name: Build and Test
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.20.3
```

Comment:
We should probably go with the latest in the 1.20 series if we're sticking with 1.20.